src/main/java/org/folio/helper/CheckinReceivePiecesHelper.java

```diff
@@ -506,11 +511,15 @@ private Future<Boolean> createHoldingsForChangedLocations(Piece piece, String in
       return Future.succeededFuture(true);
     }
 
+    String receivingTenantId = getReceivingTenantId(piece);
     String locationId = getLocationId(piece);
     String holdingId = getHoldingId(piece);
+    logger.info("createHoldingsForChangedLocations:: receivingTenantId: {} locationId: {} holdingId: {}",
+      receivingTenantId, locationId, holdingId);
     Location location = new Location().withLocationId(locationId).withHoldingId(holdingId);
-    var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, piece.getReceivingTenantId());
-    return inventoryHoldingManager.getOrCreateHoldingsRecord(instanceId, location, locationContext)
+    var locationContext = RequestContextUtil.createContextWithNewTenantId(requestContext, receivingTenantId);
+    return inventoryInstanceManager.createShadowInstanceIfNeeded(instanceId, requestContext)
```

Comment:
we need to specify locationContext here because need to created shadow instance in receiving tenant

Reply:
unit test for this new logic is missed, is it possible easily to add?

Reply:
adding..

Reply:
done